File: Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string s;
		if(n<=0) return s;
		s="1";

		int count;
		string temp;
		for(int i=2;i<=n;i++){
			temp="";
			count=1;
			for(int j=1;j<s.size();j++){
				if(s[j]==s[j-1]){
					count++;
				}
				else{
					temp+=char(count+48);
					temp+=s[j-1];
					count=1;
				}
			}
			temp+=char(count+48);
			temp+=s[s.size()-1];
			s=temp;
		}
		return s;
    }
};

File: LeetCode/count_and_say.py
class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        term = '1'
        for _ in range(n-1):
            #notice here!!!!!!!!!!!!!!
            count = 0
            curchar = term[0]
            countchar = curchar
            newterm = ''
            for j in range(0, len(term)):
                curchar = term[j]
                if j == len(term) - 1:
                    if countchar == curchar:
                        count += 1
                    else:
                        newterm += (str(count) + countchar)
                        count = 1
                    newterm += (str(count) + curchar)
                    
                else:
                    if countchar == curchar:
                        count += 1
                    else:
                        newterm += (str(count) + countchar)
                        #notice here!!!!!!!!!!!!!! should reset to 1
                        count = 1
                        countchar = curchar
            term = newterm
                        
        return term
                
                
#possible solutions
    using re
    using groupby
